src/lib/walletManager.ts

```diff
@@ -0,0 +1,182 @@
+import { JsonRpcProvider, Wallet, getAddress } from "ethers";
+import { OvalDiscovery } from "./";
+import { env } from "./env";
+import { retrieveGckmsKey } from "./gckms";
+import { isMochaTest } from "./helpers";
+import { OvalConfigs, OvalConfigsShared } from "./types";
+
+type WalletConfig = {
+    unlockerKey?: string;
+    gckmsKeyId?: string;
+};
+
+type WalletUsed = {
+    walletPubKey: string;
+    targetBlock: number;
+    count: number;
```

Comment:
Does count mean we can use a wallet more than once in a single block? Interesting... I hadn't really considered what the node would do if it ran out of wallets. I think this makes sense.